### PR TITLE
:sparkles: add support for invoice splitter auto-extraction

### PR DESCRIPTION
**Pull request by sebastianMindee**
## Description

* :sparkles: add support for invoice splitter auto-extraction
* :arrow_up: add support for integration tests
* :recycle: refactor some pdf processing code
* :recycle: cleanup pom.xml versions
* :arrow_up: bump okhttp3 to 4.12

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires a change to the official Guide documentation.

### CODE REVIEW

**Review by ianardee — CHANGES_REQUESTED** [review]

File: src/main/java/com/mindee/extraction/PDFExtractor.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  public PDFExtractor(String filePath) throws IOException {
+    this(new LocalInputSource(filePath));
+  }
+
+  /**
+   * @return The number of pages in the file.
+   */
+  public int getPageCount() {
+    return this.sourcePdf.getNumberOfPages();
+  }
```
Comment (ianardee) [inline src/main/java/com/mindee/extraction/PDFExtractor.java:44]: minor, shift this so the class constructors are grouped together

File: src/test/java/com/mindee/extraction/PDFExtractorTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    IOException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.findAndRegisterModules();
+
+    JavaType type = objectMapper.getTypeFactory().constructParametricType(
+      PredictResponse.class,
+      InvoiceSplitterV1.class
+    );
+    return objectMapper.readValue(
+      new File("src/test/resources/products/invoice_splitter/response_v1/" + "complete" + ".json"),
```
Comment (ianardee) [inline src/test/java/com/mindee/extraction/PDFExtractorTest.java:26]:
```suggestion
      new File("src/test/resources/products/invoice_splitter/response_v1/complete.json"),
```

**Review by ianardee — CHANGES_REQUESTED** [review]

File: src/test/java/com/mindee/extraction/PDFExtractorTest.java (hunk 2 above)
Comment (sebastianMindee) [reply src/test/java/com/mindee/extraction/PDFExtractorTest.java:26]: Yeah the linter still isn't fully following maven's settings, idk how to fix it :(

File: pom.xml [hunk 3]
```diff
@@ -152,7 +148,7 @@
           </configuration>
         </plugin>
       </plugins>
-    </pluginManagement>
+    </pluginManagement><plugins><plugin><groupId>org.apache.maven.plugins</groupId><artifactId>maven-compiler-plugin</artifactId><configuration><source>8</source><target>8</target></configuration></plugin></plugins>
```
Comment (ianardee) [inline pom.xml:151]: unreadable, indent properly

File: pom.xml [hunk 4]
```diff
@@ -186,6 +182,16 @@
               </execution>
             </executions>
           </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>3.3.0</version>
```
Comment (ianardee) [inline pom.xml:188]: add a variable instead of hard-coding the version

**Review by ianardee — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (sebastianMindee) [reply pom.xml:151]: IDK wtf happend there :laughing:

File: pom.xml [hunk 5]
```diff
@@ -206,6 +222,42 @@
...
+
+    <profile>
+      <id>integration-tests</id>
+      <build>
+        <plugins>
+          <!-- Integration Test Configuration -->
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-failsafe-plugin</artifactId>
+            <version>3.3.0</version>
```
Comment (ianardee) [inline pom.xml:238]: don't hardcode the version here either ;-)

**Review by ianardee — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 5 above)
Comment (sebastianMindee) [reply pom.xml:238]: Oh, wait surefire was already like that. Should I do it for every plugin then?

Comment (ianardee) [reply pom.xml:238]: just the ones that are used multiple times in the pom. The idea is to not have to hunt down all the instances of the plugin when updating it.

**Review by ianardee — APPROVED** [review]

File: .github/workflows/integration-tests.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+name: Integration Tests
+
+on:
+  pull_request:
+  workflow_run:
+    workflows: ["Test Samples Java Classes"]
```
Comment (ianardee) [inline .github/workflows/integration-tests.yml:6]: Better to allow use on all workflows I think. Just remove this line.

**Review by ianardee — CHANGES_REQUESTED** [review]

**Review by ianardee — APPROVED** [review]

**Pull request merged**